### ISSUE TITLE
/usr/bin/ld: cannot find -lboost_thread error

### ISSUE THREAD

**Issue by zufardhiyaulhaq** [zioc/contrail-devstack-plugin#51]
hi, i try to install contrail and openstack with devstack plugin. when compiling error
im using ubuntu 16.04.3 server.

> /usr/bin/ld: cannot find -lboost_thread
> collect2: error: ld returned 1 exit status
> scons: *** [build/production/vnsw/agent/ovs_tor_agent/contrail-tor-agent] Error 1
> scons: building terminated because of errors.
> +/opt/stack/contrail/devstack/plugin.sh:source:1  exit_trap
> +./stack.sh:exit_trap:515                  local r=2
> ++./stack.sh:exit_trap:516                  jobs -p
> +./stack.sh:exit_trap:516                  jobs=
> +./stack.sh:exit_trap:519                  [[ -n '' ]]
> +./stack.sh:exit_trap:525                  '[' -f '' ']'
> +./stack.sh:exit_trap:530                  kill_spinner
> +./stack.sh:kill_spinner:425               '[' '!' -z '' ']'
> +./stack.sh:exit_trap:532                  [[ 2 -ne 0 ]]
> +./stack.sh:exit_trap:533                  echo 'Error on exit'
> Error on exit
> +./stack.sh:exit_trap:535                  type -p generate-subunit
> +./stack.sh:exit_trap:536                  generate-subunit 1519890099 6089 fail
> +./stack.sh:exit_trap:538                  [[ -z /opt/stack/logs ]]
> +./stack.sh:exit_trap:541                  /home/btech/devstack/tools/worlddump.py -d /opt/stack/logs
> +./stack.sh:exit_trap:550                  exit 2
> btech@ubuntu:~/devstack$ 

anyone know how to fix this error? thanks :+1: 

before installing with ubuntu 16.04.3, i try to install with ubuntu 14.04.1, but its error.
this is the error for ubuntu 14.04.1
> E: Unable to locate package libsystemd-dev

**Comment by Doude** [comment]
Hi,
I think you hit an issue which should be resolve in the pull request #47 
Can you try it? Just need to set the that in your local.conf file:

> enable_plugin contrail https://github.com/Doude/contrail-devstack-plugin.git dev

**Comment by zufardhiyaulhaq** [comment]
i will try your suggestion. thanks

**Comment by zufardhiyaulhaq** [comment]
anyone know this issue?
```
From https://github.com/Juniper/contrail-vrouter
 * [new branch]      R1.04      -> github/R1.04
 * [new branch]      R1.05      -> github/R1.05
 * [new branch]      R1.06      -> github/R1.06
 * [new branch]      R1.06c1    -> github/R1.06c1
 * [new branch]      R1.10      -> github/R1.10
 * [new branch]      R1.30      -> github/R1.30
 * [new branch]      R2.0       -> github/R2.0
 * [new branch]      R2.1       -> github/R2.1
 * [new branch]      R2.20      -> github/R2.20
 * [new branch]      R2.20.x    -> github/R2.20.x
 * [new branch]      R2.21.x    -> github/R2.21.x
 * [new branch]      R2.22-dev  -> github/R2.22-dev
 * [new branch]      R2.22.x    -> github/R2.22.x
 * [new branch]      R3.0       -> github/R3.0
 * [new branch]      R3.0.2.x   -> github/R3.0.2.x
 * [new branch]      R3.0.3.x   -> github/R3.0.3.x
 * [new branch]      R3.1       -> github/R3.1
 * [new branch]      R3.1.1.x   -> github/R3.1.1.x
 * [new branch]      R3.2       -> github/R3.2
 * [new branch]      R3.2.3.x   -> github/R3.2.3.x
 * [new branch]      R4.0       -> github/R4.0
 * [new branch]      R4.1       -> github/R4.1
 * [new branch]      master     -> github/master
 * [new tag]         v1.05      -> v1.05
 * [new tag]         v1.06      -> v1.06
 * [new tag]         v1.10      -> v1.10
 * [new tag]         v1.20      -> v1.20
 * [new tag]         v1.21      -> v1.21
 * [new tag]         v2.0       -> v2.0
 * [new tag]         v2.01      -> v2.01
 * [new tag]         v2.10      -> v2.10
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
```

**Comment by Doude** [comment]
Can you try to run `repo sync` in contrail source folder or set `RECLONE` to true in your devstack config file?

**Comment by zufardhiyaulhaq** [comment]
in the stack.sh? i cant find variable ```RECLONE```. by the way, thanks for answer. its works now. but its get another error.
```
Install file: "build/production/config/fabric-ansible/ansible-playbooks/dist/fabric_ansible_playbooks-0.1dev.tar.gz" as "/usr/bin/fabric_ansible_playbooks-0.1dev.tar.gz"
scons: *** [/usr/bin/fabric_ansible_playbooks-0.1dev.tar.gz] build/production/config/fabric-ansible/ansible-playbooks/dist/fabric_ansible_playbooks-0.1dev.tar.gz: No such file or directory
scons: building terminated because of errors.
+/opt/stack/contrail/devstack/plugin.sh:source:1  exit_trap
+./stack.sh:exit_trap:515                  local r=2
++./stack.sh:exit_trap:516                  jobs -p
+./stack.sh:exit_trap:516                  jobs=
+./stack.sh:exit_trap:519                  [[ -n '' ]]
+./stack.sh:exit_trap:525                  '[' -f '' ']'
+./stack.sh:exit_trap:530                  kill_spinner
+./stack.sh:kill_spinner:425               '[' '!' -z '' ']'
+./stack.sh:exit_trap:532                  [[ 2 -ne 0 ]]
+./stack.sh:exit_trap:533                  echo 'Error on exit'
Error on exit
+./stack.sh:exit_trap:535                  type -p generate-subunit
+./stack.sh:exit_trap:536                  generate-subunit 1520333256 8478 fail
+./stack.sh:exit_trap:538                  [[ -z /opt/stack/logs ]]
+./stack.sh:exit_trap:541                  /home/btech/devstack/tools/worlddump.py -d /opt/stack/logs
```

**Comment by Doude** [comment]
No in `local.conf` file, this is the only one devstack config file. `stack.sh` is the devstack script, don't edit it if you don't what you're doing.

For that new issue you had, it the first time I saw it. Probably a new repository in contrail. Lot of changes are done in the Contrail source repositories actually on master branches. Probably the reclone tips I gave you will fix that.

**Comment by Doude** [comment]
Do you

**Comment by Doude** [comment]
Fixed by #47
